File: wc-app-back/world-cup-app/src/main/java/com/wc_app_back/world/cup/app/entities/YearlySummary.java
package com.wc_app_back.world.cup.app.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class YearlySummary {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Integer year;

    private String host;

    private String champion;

    private String runnerUp;

    private Integer matches;

    private String topScorerPlayer;

    private Integer topScorerGoals;
}

File: wc-app-ai/world_cup_tournaments/pipeline/yearly_summary.py
from pipeline.preprocessing import split_top_scorer


def yearly_summary(df):

    df = split_top_scorer(df.copy())

    return df[[
        "Year",
        "Host",
        "Champion",
        "Runner-Up",
        "Matches",
        "TopScorerPlayer",
        "TopScorerGoals"
    ]]